### DISCUSSION TITLE
Incorrect threshold value

### DISCUSSION THREAD

**Discussion by DheerajMaruthi** [aurelio-labs/semantic-router#450] [General]
Threshold values are vague and changes everytime we train with same data. Anybody else facing this issue?

**Comment by jamescalam** [comment]
They are not deterministic, so it is expected that they will change — although the optimal accuracy should remain roughly the same between runs

**Reply by DheerajMaruthi** [reply]
No they are not same, some times thresholds are as low as .2 and as high as .9

This is extremely random
